File: src/python/ch01/1.1/l0/dec_to_bin.py
def dec_to_bin(n: int) -> str:
    """Convert a non-negative integer to binary string (no prefix)."""
    if n < 0:
        raise ValueError("Only non-negative integers supported.")
    if n == 0:
        return "0"

    bits = []
    while n > 0:
        bits.append(str(n % 2))
        n //= 2
    return "".join(reversed(bits))

def main():
    try:
        text = input("Enter a non-negative integer: ").strip()
        n = int(text)
        print(dec_to_bin(n))
    except ValueError:
        print("Invalid input.")

if __name__ == "__main__":
    main()

File: src/erlang/ch01/1.1/l0/src/dec_to_bin.erl
-module(dec_to_bin).
-export([convert/1]).

%% Convert a non-negative integer to its binary string (as list of chars).
convert(0) -> "0";
convert(N) when is_integer(N), N >= 0 ->
    convert(N, []).
    
convert(0, Acc) -> Acc;
convert(N, Acc) ->
    Bit = integer_to_list(N rem 2),
    convert(N div 2, [Bit | Acc]).
